site/docs/components/_content-template/index.mdx

```diff
@@ -0,0 +1,213 @@
+    },
+    // ...
+  ],
+};
+
+# Component Name
+
+<!--
+  Example screenshot or live component demo here of the default/most common
+  usage.
+
+  For screenshots use Markdown:
+
+  ![Text alternative for screenshot](./path/to/screenshot.jpg)
+
+  For live demos in Storybook use:
+
+  <Canvas>
+    <Story id="..." />
+  </Canvas>
+-->
+
+<!--
+  Introductory text that briefly explains the purpose of the component and
+  anything else of note.
+-->
+
+## When to use Component Name
+
+<!-- Bulleted list of scenarios in which this component can be used  -->
+
+- ...
+- ...
+
+## When not to use Component Name
+
+<!--
+  Bulleted list of scenarios where this component is not appropriate. Where
+  possible, include a link to an alternative component that should be used in
+  it's place
+-->
+
+- ...
+- ...
+
+## Configurations
+
+<!--
+  A visual list of possible configurations of the component.
+
+  The purpose is to convey the main features and/or usage options of the
+  component (e.g. the Data Grid's striped rows might be a "configuration" to
+  showcase here). The configurations listed here should aim to be comprehensive,
+  but not necessarily exhaustive (some complex components may have vast numbers
+  of possible permutations, so listing out each one would be impossible and
+  overwhelming to readers).
+
+  Each confguration should have its own section following the structure outlined
+  below:
+-->
+
+### Configuration title (e.g. Striped rows)
+
+<!--
+  A brief description of what this configuration provides to users and when it
+  should be used.
+-->
+
+<!--
+  A live example of the component configuration, with access to the correspoding
+  source code.
+
+  In Storybook this is done using something like:
+
+  <Canvas>
+    <Story id="..." />
+  </Canvas>
+-->
+
+## Content
+
+<!--
+  OPTIONAL: Remove this section if it is not applicable.
+
+  Add any component-specific content guidance here, use visual examples where
+  appropriate.
+
+  For example, a Button component might include guidance on recommended terms to
+  use for the label text, recommended maximum character counts, etc.
+-->
+
+## Accessibility
+
+<!--
+  OPTIONAL: Remove this section if it is not applicable.
+
+  Describe any considerations the users may need to take into account when using
+  the component.
+-->
+
+## References
+
+<!--
+  OPTIONAL: Remove this section if it is not applicable.
+
+  If there are industry standards for this component, list them in this
+  section. Include links out to useful articles.
+-->
+
+- [Relevant industry standard](https://example-standard.org/)
+- [Useful article](https://some-site.com/article/bla)
+- ...
+
+## Designing
+
+### Theming considerations
+
+<!--
+  Describe how Salt characteristics and theming impact the component, and any
+  modifications that can be made.
+-->
+
+### Design guidance
+
+<!--
+  Add a bulleted list of any other design-related guidance the user may need to
+  consider.
```

Comment:
Would be nice if we can provide a short example of the kind of guidance that would included here. Bear in mind we already explain the component's purpose, dos and don'ts further up the page - so what _else_ is it that would go here?

@shey-v, @bhoppers2008 if you have suggestions we could include here, that would be awesome!

Also, might this section include Figma component usage instructions (e.g. if there are non-intuitive quirks designers need to be aware of in order to successfully use this component in Figma)? Or would that kind of thing deserve a new (optional) section of its own?

Reply:
I'm keen to move do's and don'ts (when to use and when not to use) down here as it is specific to design usage, patterns and the component behaviour. 

i.e. https://spectrum.adobe.com/page/button/#Usage-guidelines
i.e. https://atlassian.design/components/button/usage#best-practices

This kind of content would be relevant here. 

From what I've seen, each component in the Community space and internally, comes with Figma component guides which tell the user how to configure the component to the variants and states they need etc. I think perhaps this could go into it's own section or just be mentioned in the copy, i.e. refer to the component guides to view Fgima component usage instructions?

Reply:
I'd prefer not to move this stuff under a section labelled "Design(ing)" as it's relevant information for product & dev people too. My worry is that people in those roles wouldn't think to look at that section/tab since they don't indentify as designers.

I therefore propose we create a "Usage" (or "Guidelines" or something else to that effect) section and move the Dos & Don'ts and also this references section in there. Hopefully that would be perceived as more inclusive and so not only designers, but others will look at this too.

The Atlassian example you linked to does this. Others like IBM Carbon & Orbit Kiwi also do this and I really like it:

![screenshot of a Carbon component page featuring a "Usage" tab](https://user-images.githubusercontent.com/16718916/219608833-834691b3-2fc0-42a8-bab3-2a1afa304cdd.png)

![screenshot of a Kiwi component page featuring a "Guidelines" tab](https://user-images.githubusercontent.com/16718916/219609114-990a353c-7991-4777-93ab-ef2a2a2310ba.png)

Reply:
Fair enough, I'll have to change my designs.

Reply:
I do think this info is particularly useful for users who aren't designers. I'm happy to have another heading but I'm not fond of the word 'Usage' as it's a bit vague. The whole page is about Usage really.

Reply:
OK. I've pushed an update that does what I suggested (I went with "Usage" as the title for now, but happy to change that if others have better suggestions).

@sion-docs & @shey-v please take a look and let me know what you think. Cheers!

Reply:
I was thinking about seperating this info out into 'when to use' and 'when not to use'. I feel like perhaps we need an 'Overview' tab, where this info lives, followed by 'Developing' then 'Designing' maybe?

Reply:
@sion-docs  @james-nash  In this section Designers would like to see:

* Best practises and high level guidance in terms of design principles.

* An anatomy diagram. This would be helpful for designers to clearly communicate what a component is made up of - particularly for components that are made up of a set of different components. I.e. a Form Field. + making it clear that an input is inside of a FormField and within that is an icon… etc. 

* A link to a sticker sheet. This would be helpful for designers to see ALL variations and permutations that are possible with the components, i.e. our current FF documentation page doesn’t show you all examples and possible alignments - label left for example 

* Sizing and spacing spec. This would be useful when designers who aren’t familiar with our densities and base X values… and considering there will be a new size of ultra hd in the release of the data grid component.. it would be a nice to have.

Reply:
Does it actually help us to create silo'd views of these component pages. I'm beginning to think it makes more sense to separate content by intent, not subject. Using, Styling, Coding, etc. What we think might be of interest to a designer, might also be useful to dev, and vice versa. Everything Shey has linked to designing, could also fit under a 'Style' section, or spill into 'Usage', or even touch on 'code'. This overlaps with the conversation we had about design tokens yesterday... that's something yet to be defined, but would be good to agree the structure now, that would easily accommodate something related to 'tokens', when made 'visible'.

Reply:
💯 on using intent rather than subject or role. I'll have a stab at renaming / rejigging things in moment.

Reply:
Discussed with @shey-v on a call and we've agreed to reintroduce the Anatomy section as all research so far indicates there's a strong demand for that from designers.

Leaving away the sizing/spacing section for now though, since we need to better understand _why_ people want that. If we get sufficient clarity, we can add that in a future revision of this template.